File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.c
int maximumCount(int* nums, int numsSize) {
    
    int ng=-1;
    int low=0,high=numsSize-1;

    while(low<=high){
        int mid=low+(high-low)/2;

        if(nums[mid]<0){
            ng=mid;
            low=mid+1;
        }
        else{
            high=mid-1;
        }
    }
    int ncount=ng+1;


 int po=-1;
    low=0,high=numsSize-1;

    while(low<=high){
        int mid=low+(high-low)/2;

        if(nums[mid]>0){
            po=mid;
            high=mid-1;
        }
        else{
            low=mid+1;
        }
    }

    int pcount;
    if(po!=-1)
        pcount=numsSize-po;
    else
        pcount=0;

    return (pcount>ncount)?pcount:ncount;
}


/*

int pcount=0,ncount=0;
    for(int index=0;index<numsSize;index++){
        if(nums[index]>0){
            pcount++;
        }
        else if(nums[index]<0){
            ncount++;
        }
    }

    return (pcount>ncount)?pcount:ncount;
*/

File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.java
class Solution {
    public int maximumCount(int[] nums) {
        
        //last occurance of neg number
        int low=0,high=nums.length-1;
        int neg_index=-1;

        while(low<=high){
            int mid=low+(high-low)/2;

            if(nums[mid]<0){
                neg_index=mid;
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        int neg_count=neg_index+1;
        

         //1st occurance of pos number
        low=0;
        high=nums.length-1;
        int pos_index=-1;

        while(low<=high){
            int mid=low+(high-low)/2;

            if(nums[mid]>0){
                pos_index=mid;
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        int pos_count;
        if(pos_index!=-1){
            pos_count=nums.length-pos_index;
        }
        else{
            pos_count=0;
        }

        System.out.println("Neg_index: "+neg_index+" posindex "+pos_index);
        System.out.println("Neg: "+neg_count+" pos: "+pos_count);
        return Math.max(pos_count,neg_count);

    }
}